jupyterhub_moss/spawner.py

```diff
@@ -358,20 +367,27 @@ def options_from_form(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
             # Set path to use from first environment for the current partition
             options["environment_path"] = partition_environments[0]["path"]
 
-        # Singularity images are never added to PATH
-        if options["environment_path"].endswith(".sif"):
-            return options
-
-        # Custom envs are always added to PATH, defaults ones only if add_to_path is True
         corresponding_default_env = find(
             lambda env: env["path"] == options["environment_path"],
             partition_environments,
         )
-        if (
+
+        # Generate prologue
+        prologue = self.req_prologue
```

Comment:
This traitlet from `batchspawner` was not taken into account before, best to handle it.